node/src/components/consensus/highway_core/finality_detector.rs

```
@@ -61,7 +65,7 @@ impl<C: Context> FinalityDetector<C> {
         }
         Ok(iter::from_fn(move || {
             let bhash = self.next_finalized(state)?;
-            // Safe to unwrap: Index exists, since we have units from them.
+            // NOTE: Safe to unwrap: Index exists, since we have units from them.
             let to_id = |vidx: ValidatorIndex| highway.validators().id(vidx).unwrap().clone();
```

Seems like we could return an error here.  @fizyk20 any reason to not do that?